src/librustdoc/html/static/js/settings.js

```diff
@@ -88,10 +91,183 @@
+                        <div class="choices">`;
+                // output += `<div>${setting["name"]}</div>
+                //     <label class="select-wrapper">
+                //         <select id="${setting["js_name"]}" autocomplete="off">`;
+                onEach(setting["options"], function(option) {
+                    var selected = option === setting["default"] ? " selected" : "";
+
+                    output += `<label for="${js_data_name}-${option}" class="choice">\
+                           <input type="radio" name="${js_data_name}" \
+                                id="${js_data_name}-${option}" value="${option}"${selected}>\
+                           ${option}\
+                         </label>`;
+                    // output += `<option value="${option}"${selected}>${option}</option>`;
+                });
+                // output += `</select>
+                //     <img src="${resourcePath("down-arrow", ".svg")}" alt="Select item">
+                // </label>`;
+                output += "</div></div>";
+            } else {
+                // This is a toggle.
+                var checked = setting["default"] === true ? " checked" : "";
+                output += `
+                    <label class="toggle">
+                        <input type="checkbox" id="${setting["js_name"]}"${checked}>
+                        <span class="slider"></span>
+                    </label>
+                    <div>${setting["name"]}</div>`;
+            }
+            output += "</div>";
         });
+        return output;
+    }
+
+    function buildSettingsPage(settings) {
+        // First, we add the settings.css file.
+        var link = document.createElement("link");
+        link.href = resourcePath("settings", ".css");
+        link.type = "text/css";
+        link.rel = "stylesheet";
+        document.getElementsByTagName("head")[0].appendChild(link);
+
+        // Then we build the DOM.
+        var el = document.createElement("section");
+        el.id = "settings";
+        el.innerHTML = `
+            <h1 class="fqn">
+                <span class="in-band">Rustdoc settings</span>
+            </h1>
+            <div class="settings">${buildSettings(settings)}</div>`;
+
+        if (isSettingsPage) {
+            document.getElementById(MAIN_ID).appendChild(el);
+        } else {
+            var closeButton = document.createElement("button");
+            closeButton.className = "close-button";
+            closeButton.textContent = "Close settings";
+            closeButton.onclick = function() {
+                switchDisplayedElement(null);
+            };
+            el.appendChild(closeButton);
+
+            getNotDisplayedElem().appendChild(el);
+        }
+        return el;
+    }
+
+    function createSettingsPage() {
+        var themes = getVar("themes").split(",");
+        var settings = [
+            {
+                "name": "Theme preferences",
```

Comment:
The "Theme preferences" heading (and horizontal line under it) have been removed in the latest nightly; we should remove it here too.